prosessering-core/src/main/kotlin/no/nav/familie/prosessering/external/TaskService.kt

```diff
@@ -0,0 +1,25 @@
+package no.nav.familie.prosessering.external
+
+import no.nav.familie.prosessering.domene.ITask
+import no.nav.familie.prosessering.domene.TaskRepository
+import no.nav.familie.prosessering.internal.TaskStepExecutorService
+
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.core.task.TaskExecutor
+import org.springframework.stereotype.Component
+
+@Component
+class TaskService(val taskRepository: TaskRepository,
+                  @Qualifier("taskExecutor") private val taskExecutor: TaskExecutor,
+                  private val taskStepExecutorService: TaskStepExecutorService)  {
+
+    fun save(task: ITask): ITask {
+        return taskRepository.save(task)
+    }
+
+    fun saveAndRun(task: ITask) {
+        taskRepository.save(task)
+
+        taskExecutor.execute { taskStepExecutorService.executeWork(task) }
```

Comment:
save her burde kanskje lagres i en egen transaskjon? 
```suggestion
        val lagretTask = taskRepository.save(task)

        taskExecutor.execute { taskStepExecutorService.executeWork(lagretTask) }
```

Reply:
Bra observasjoner, laget en test også for å verifisere at dette ikke virker. pga `@Transactional(propagation = Propagation.REQUIRES_NEW)` i taskWorker...

Reply:
Ja pga taskRepository.save er fortsatt i en transaksjon, mens executeWork -> markerPlukket er i en ny transaksjon slik att den då ikke kan finne den tasken som ble lagret. 
Ikke helt sikker på hva som er riktig vei å gå her. Man skulle kunne opprette `saveInNewTransaction` i InternalTaskService og la TaskService kalle den når man oppretter en ny task.

Samtidig, hvis noe skjer etter att man lagret tasken som ruller tilbake allt annet i transaksjonen, så kommer man ha opprettet en ny task, men rullet tilbake allt. Nesten lit skummelt med slik oppførsel :P 

Lurer nesten på om det er enklere å la hver klient opprette en "saveAndRun" slik att man får den oppførselen som man ønsker?